AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -46,12 +66,36 @@ public struct GraphQLFilterConverter {
 
 /// Extension to translate a `QueryPredicate` into a GraphQL query variables object
 extension QueryPredicate {
+    public func getGraphQLFilter(_ modelSchema: ModelSchema) -> GraphQLFilter {
+        if let operation = self as? QueryPredicateOperation {
+            return operation.graphQLFilterInternal(modelSchema)
+        } else if let group = self as? QueryPredicateGroup {
+            return group.graphQLFilterInternal(modelSchema)
+        }
 
+        preconditionFailure(
+            "Could not find QueryPredicateOperation or QueryPredicateGroup for \(String(describing: self))")
+    }
+
+    @available(*, deprecated, message: """
+    Use `graphQLFilter(_)` instead. See https://github.com/aws-amplify/amplify-ios/pull/965 for more details.
+    """)
     public var graphQLFilter: GraphQLFilter {
         if let operation = self as? QueryPredicateOperation {
-            return operation.graphQLFilter
+            return operation.graphQLFilterInternal()
+        } else if let group = self as? QueryPredicateGroup {
+            return group.graphQLFilterInternal()
+        }
+
+        preconditionFailure(
+            "Could not find QueryPredicateOperation or QueryPredicateGroup for \(String(describing: self))")
+    }
```

Comment:
Is there a reason for keeping the `graphQLFliter()` around as a deprecated method that can work in some cases (and fail in others)?  Seems like we can just change the signature and force a minor version update?

Reply:
Would removing a public API that has a bug for another API be major version bump? I believe if it is a major version bump, we should maintain the deprecated API for now and do the major version bump later (2.0.0), but if it is just a minor version then we should do it now